### DISCUSSION TITLE
Event on menu creation

### DISCUSSION THREAD

**Discussion by andrew-kamenev** [GeorgianStan/vanilla-context-menu#40] [Q&A]
Hello, is there any event which triggers when menu is created? I want to add custom element in menu... Thanks

**Comment by andrew-kamenev** [comment]
Not actual, found solution

**Reply by GeorgianStan** [reply]
Hi @andrew-kamenev did you manage to figure it out? You can use `updateOptions` do change the options of the context menu.

**Reply by andrew-kamenev** [reply]
I have another problem
Width of menu always 150px, 
when I change it with css, for example to 230px, position of menu calculates from default width (150px), 
so when i click on the right side of the screen menu is cut

**Reply by GeorgianStan** [reply]
Hi! Yes, you are right! Can you create an issue for this please?

**Reply by stko** [reply]
> Not actual, found solution

What's your solution? I just start to try out vanillaContextMenu for a drawing application where I would need to adjust the menu all the time depending on the actual state of the selected element..
